README.md
# BuildBox

The repository provides a complete set of build tools for web developpers. **These
images MUST NOT be used in production**. The targeted usage of those images is GitlabCI.

## Available images

### AWS

Contains AWS Cli + CI Helper

### DIND - AWS

Adds AWS Cli & CI Helper to GitLab's dind image (to run docker in a GitLab runner).

Use case:
```yaml
# .gitlab-ci.yml
test:
  image: ekino/docker-buildbox:latest-dind-aws
  services:
    - ekino/docker-buildbox:latest-dind-aws
  variables:
    DOCKER_DRIVER: overlay2
    DOCKER_HOST:   "tcp://ekino__docker-buildbox:2375"
  script:
    - docker ...
```

### Golang

Based upon official Golang image, contains glide, gin, AWS Cli and CI Helper.

### Java

Contains AWS Cli, CI Helper, and Java 8.

### Node

Contains node (installed in the NODE_VERSION env var value), CI Helper and AWS Cli.

### PHP

Contains PHP (installed from official alpine in the PHP_VERSION env var value) within Blackfire, Composer, PHP CS Fixer, Security Checker, AWS Cli and CI Helper.

About Blackfire, please read the official documentation to install the agent https://blackfire.io/docs/integrations/docker, then you should be able to profile a PHP script like this:

```bash
docker exec -it -e BLACKFIRE_CLIENT_ID -e BLACKFIRE_CLIENT_TOKEN my-php-container blackfire run bin/console app:foo:bar
```

### Ruby

Contains Ruby (installed from official alpine) and CI Helper.

### Ansible

Contains Ansible, CI Helper and Python 2.7

### SonarQube Scanner

Contains SonarQube Scanner and CI Helper

### Sonarts
Contains node (installed in the NODE_VERSION env var value), SonarQube Scanner and CI Helper
SonarTs plugin need local node to send information to sonar.

### React Native Android

Contains Java 8, Gradle, Android SDK, Node 7.10 and React Native Cli

## Versions

Please review the [CHANGELOG.md](CHANGELOG.md) file for versions per tag.

## Testing

Each box is tested and built using TravisCI.

The `travis.py` script try to be clever:
 - PR: only images with modified files are built.
 - Merge to master: only images with modified files are built and pushed to the docker registry with the tag `latest-IMAGE`
 - TAG: all images are built and pushed to the docker registry
 - Nightly: all images are built and pushed to the docker registry with the tag `nightly-IMAGE`

[![Build Status](https://travis-ci.org/ekino/docker-buildbox.svg?branch=master)](https://travis-ci.org/ekino/docker-buildbox)

It is possible to build local image for testing with the following command:

    CI_HELPER_VERSION=0.0.5 MODD_VERSION=0.5 JAVA_VERSION=10.0.1-jdk-slim VERSION=10 python travis.py --language java --pull-request=true


CHANGELOG.md
Versions
========

2018-07-DEV
-----------

* Upgrade ci-helper: 0.0.6
* Add Blackfire probe & client in PHP images
* Use openjdk slim as base Java image
* Add Java version 10.0.1
* Use openjdk slim as base React Native image
* Remove Maven and Gradle from React Native image
* Add bash to Python images
* Add an up to date mime types definition file for the images that use AWS CLI
* Install PEAR in the PHP 5.3 image
* Add sonarts image

2018-06-14
----------

* Add rsync in PHP images
* Add exif extension in PHP images
* Add PHP 7.2.6
* Upgrade PHP version: 7.1.18
* Upgrade Node version: 6.14.3, 8.11.3, 10.4.1
* Upgrade Sonar Scanner: 3.2.0.1227
* Upgrade ci-helper: 0.0.5

2018-05-18
----------

* Upgrade Docker version to 18.05.0 and Docker Compose to 1.21.2 in dind-aws image
* Upgrade Node version: 6.14.2, 8.11.2, 9.11.1
* Upgrade PHP version: 5.6.36, 7.1.17
* Upgrade Composer version: 1.6.5
* Upgrade PHP Redis version: 4.0.2
* Upgrade Ruby version: 2.5.1
* Add Python images: versions 2.7 and 3.6
* Add tests on PHP images
* Add React Native image
* Upgrade Java version: 8u171-1\~webupd8\~0
* Fix intl in PHP images

2018-04-05
----------

* Upgrade Python version in Ansible image: 2.7.14-r0
* Upgrade Docker version to 18.03.0 and Docker Compose to 1.20.1 in dind-aws image
* Upgrade Node version: 6.14.1, 8.11.1, 9.10.1
* Upgrade PHP version: 5.6.35, 7.1.16
* Upgrade APCu version for PHP 7.1: 5.1.11
* Upgrade Composer version: 1.6.3
* Upgrade PHP Redis version: 4.0.0
* Upgrade Sensiolabs' security-checker version for PHP 7: 4.1.8
* Upgrade Xdebug version for PHP 7: 2.6.0
* Use Alpine binary of ci-helper for Ansible, AWS, DIND-AWS, PHP >= 5.6, Ruby and Sonar images
* Fix deprecated MAINTAINER instruction
* BC break: PHP image is now based on Alpine
    - intl for PHP 7.2 is not released yet, so image 7.2 is not available anymore
* Upgrade modd version: 0.5

2018-01-25
----------

* Upgrade NPM version in Node image (fix node@9 incompatibility):
    - node v8.9.4  -> npm v5.6.0
    - node v9.4.0  -> npm v5.6.0
* Upgrade ci-helper version: 0.0.4
* Upgrade Docker version to 18.01.0 and Docker Compose to 1.18.0 in dind-aws image
* Upgrade Java version: 8u161-1\~webupd8\~0
* Upgrade Node version: 6.12.3, 8.9.4, 9.4.0
* Upgrade APCu version: 5.1.9
* Upgrade Composer version: 1.6.2
* Upgrade PHP Redis version: 3.1.6
* Upgrade Sensiolabs' security-checker version for PHP 7: 4.1.7
* Upgrade PHP version: 5.6.33, 7.2.1
* Upgrade Ruby version: 2.5.0
* Add rsync to aws image
* Add zip to aws image
* Add postgresql-client to PHP image

2017-12-12
----------

* Add tzdata into sonar image
* Upgrade Ruby version: 2.4.2
* Upgrade Glide version: 0.13.1
* Upgrade Docker Compose version: 1.17.1
* Upgrade Composer version: 1.5.5
* Upgrade PHP version: 5.6.32, 7.1.12
* Upgrade Node version: 6.12.2, 8.9.3, 9.2.1
* Allow specifying NPM version in Node image :
    - node v6.11.3 -> npm v3.10.10
    - node v8.9.1  -> npm v5.5.1
    - node v9.2.0  -> npm v5.5.1
* Add rsync and tzdata into dind-aws image
* Add Sensiolabs' security-checker to PHP images
* Add make to aws image

2017-11-01
----------

* BC break: dind-aws image is now based on Alpine
* Upgrade Docker version to 17.09 and Docker Compose to 1.16.1 in dind-aws image
* Upgrade Java version: 8u151-1\~webupd8\~0
* Upgrade PHP version: 5.6.31, 7.1.9
* Upgrade PHP Redis: 3.1.4
* Add Node 8.6
* Upgrade node 6.11.3
* Remove Node 4.8
* Upgrade golang: 1.9
* Upgrade Glide: 0.13.0
* Add Ansible: 2.2.3.0
* Add Sonar Scanner: 3.0.3.778

2017-07-17
----------

* Upgrade PhpRedis: 3.1.3
* Add a "new" PHP image (5.3) for legacy support
* Upgrade Node version: 4.8.4, 6.11.1, 7.10.1
* Upgrade PHP version: 7.1.7
* Upgrade composer version: 1.4.2
* Upgrade the base Debian image: 8.7
* Upgrade Java version: 8u131
* Add imagemagick, gcc and g++ into Node image
* Add a new Ruby image
* Add composer hirak/prestissimo plugin
* Upgrade go version: 1.8

2017-03-30
----------

* Upgrade Java version: 8u121-1\~webupd8\~2
* Add PHP extension: intl
* Add bzip2 into Node, Java and PHP images
* Upgrade ci-helper to version 0.0.3 to support gitlab 9.x
* Update PHP version: 7.1.3
* Update Node version: 4.8.1, 6.10.1, 7.8.0

2017-02-07
----------

* Add PHP extension for Redis
* Add PHP version: 7.1.1
* Remove PHP version: 7.0.14
* Upgrade PHP version: 5.6.30
* Upgrade APCu version: 5.1.8
* Upgrade composer version: 1.3.2
* Upgrade Java version: 8u121
* Upgrade Maven version: 3.3.9
* Upgrade Node version: 4.7.3, 6.9.5, 7.5.0
* Add a new Golang image

2017-01-06
----------

* Add a new AWS image with awscli and python libs
* Add CI-Helper 0.0.2 into base image
* Add a new DIND AWS image with AWS Cli
* Add Modd 0.0.4 into base image
* Upgrade Node version: 4.7.2, 6.9.4, 7.4.0
* Upgrade PHP version: 5.6.29, 7.0.14
* Upgrade composer version: 1.3.0

2016.12-19
----------

* Slightly reduce the size of the Java image
* Add the AWS CLI to all images
* Add packages ocaml & libelf-dev for FlowType (Node) checking
* Upgrade Node versions: 4.7.0, 7.2.1
* Upgrade PHP: 7.0.14
* Fix Java Version to 8u111+8u111arm-1~webupd8~0
* Fix composer version to 1.2.4
* Migrate builds and tests to TravisCI, add support for nightly build
* Tweak PHP build configuration

2016.12.05
----------

* Fix maven install on Java
* Fix clean commands
* Update Yarn 0.18

2016.11.25
----------

* Use Debian 8.6 as base image
* Upgrade PHP versions: 5.6.28, 7.0.13
* Add driver pdo_pgsql in PHP images
* Add cache APCu in PHP images
* Upgrade Node versions: 4.6.2, 6.9.1 (LTS)
* Add Node 7.0.0
* Add maven to java8
* Remove global packages install from node: gulp, bower, grunt-cli, webpack, browserify, babel, eslint, eslint-plugin-react, eslint-plugin-angular, eslint-config-standard, eslint-plugin-promise, eslint-plugin-standard.
* Add Yarn to node
* Update nvm to 0.32.1

2016.09.16
----------

* Use Debian 8.5 as base image
* Upgrade PHP versions: 5.6.22, 7.0.10
* Remove node versions: 4.3, 6.0, 5.7
* Add node version: 6.5

travis.py
import os, subprocess, sys, argparse
import collections
import re

BuildInfo = collections.namedtuple('BuildInfo', 'language version ci_helper_version commit_range branch tag pull_request event_type is_travis modd')

def main():
    parser = argparse.ArgumentParser(description='Build some docker images.')
    parser.add_argument('--language', dest='language', default=os.environ.get('LANGUAGE'), help='The language/image to build')
    parser.add_argument('--version', dest='version', default=os.environ.get('VERSION'), help='The language\'s version to build')
    parser.add_argument('--ci-helper-version', dest='ci_helper_version', default=os.environ.get('CI_HELPER_VERSION', '0.0.6'), help='the ci helper version')
    parser.add_argument('--commit-range', dest='commit_range', default=os.environ.get('TRAVIS_COMMIT_RANGE', 'HEAD...HEAD'), help='the commit range')
    parser.add_argument('--branch', dest='branch', default=os.environ.get('TRAVIS_BRANCH', ""), help='the commit branch')
    parser.add_argument('--tag', dest='tag', default=os.environ.get('TRAVIS_TAG', False), help='the commit tag')
    parser.add_argument('--pull-request', dest='pull_request', default=os.environ.get('TRAVIS_PULL_REQUEST', "false"), help='it is a PR?')
    parser.add_argument('--event-type', dest='event_type', default=os.environ.get('TRAVIS_EVENT_TYPE', ""), help='The event type which trigger the build')
    parser.add_argument('--travis', dest='travis', default=os.environ.get('TRAVIS', "false"), help='is travis')
    parser.add_argument('--modd', dest='modd', default='0.5', help='is travis')

    args = parser.parse_args()

    buildInfo = BuildInfo(
        language=args.language,
        version=args.version,
        ci_helper_version=args.ci_helper_version,
        commit_range=args.commit_range,
        branch=args.branch if len(args.branch) > 0 else False,
        tag=args.tag,
        pull_request=True if args.pull_request != "false" else False,
        event_type=args.event_type,
        is_travis=True if args.travis == "true" else False,
        modd=args.modd,
    )

    print args, buildInfo

    run_build(buildInfo)


def run_command_exit(command, exit_message):
    if run_command(command) != 0:
        print exit_message
        sys.exit(1)


def run_command(command):
    process = subprocess.Popen(command, stdout=subprocess.PIPE, shell=True)
    while True:
        output = process.stdout.readline()
        if output == '' and process.poll() is not None:
            break
        if output:
            print output.strip()

    rc = process.poll()

    return rc


def run_build(buildInfo):
    if not buildInfo.language or not buildInfo.version:
        print "please provide a LANGUAGE and a VERSION"
        sys.exit(1)

    commit_range = buildInfo.commit_range.replace("...", "..")
    print "git diff --name-only %s | sort | uniq" % commit_range

    files = subprocess.Popen("git diff --name-only %s | sort | uniq" % commit_range, stdout=subprocess.PIPE, shell=True).stdout.read()
    language = buildInfo.language
    version = buildInfo.version
    ci_helper_version = buildInfo.ci_helper_version
    base_image = "%s%s" % (language, version)

    if language in ["aws", "dind-aws"]:
        base_image = language

    is_tag = False
    is_pr = False
    is_release = False
    is_master = False
    start_build = False
    push_image = False

    print "TRAVIS_BRANCH=%s" % buildInfo.branch
    print "TRAVIS_TAG=%s" % buildInfo.tag
    print "TRAVIS_PULL_REQUEST=%s" % buildInfo.pull_request
    print "LANGUAGE=%s" % language
    print "VERSION=%s" % version
    print "CI_HELPER_VERSION=%s" % buildInfo.ci_helper_version
    print "TRAVIS_COMMIT_RANGE=%s" % commit_range

    print "BASE IMAGE: %s" % base_image
    print "MODIFIED FILES: %s" % files

    if buildInfo.pull_request:
        print " > This is a PR"
        is_pr = True

    if buildInfo.tag:
        print " > This is a Tag"
        is_tag = True
        image = "ekino/docker-buildbox:%s-%s" % (base_image, buildInfo.tag)
        start_build = True  # on tag build all images
        push_image = True
    elif buildInfo.branch == 'master' and not is_pr:
        print " > This is a master commit"
        is_master = True
        image = "ekino/docker-buildbox:latest-%s" % (base_image)
    else:
        image = base_image

    if buildInfo.event_type == "cron":
        print " > This is a cron event"
        image = "ekino/docker-buildbox:nightly-%s" % (base_image)
        start_build = True
        push_image = True

    if is_pr and is_tag:
        print "cannot be a tag and a pr"
        sys.exit(1)

    if (is_pr or is_master) and (not buildInfo.is_travis or language in files or re.search(r".travis.yml|travis.py", files)):
        start_build = True
        push_image = is_master

    print "is_pr: %s" % is_pr
    print "is_tag: %s" % is_tag
    print "is_master: %s" % is_master
    print "is_release: %s" % is_release
    print "image: %s" % image
    print "start_build: %s" % start_build
    print "push_image: %s" % push_image

    if start_build:
        build_args = "--build-arg CI_HELPER_VERSION=%s" % ci_helper_version
        build_context = language
        run_args = "--rm"

        if language == "ansible":
            build_args = "%s --build-arg VERSION=%s --build-arg PYTHON_VERSION=%s" % (build_args, buildInfo.version, os.environ.get("PYTHON_VERSION"))

        if language == "dind-aws":
            build_args = "%s --build-arg DOCKER_COMPOSE_VERSION=%s --build-arg GLIBC_VERSION=%s" % (build_args, os.environ.get("DOCKER_COMPOSE_VERSION"), os.environ.get("GLIBC_VERSION"))

        if language == "golang":
            build_args = "%s --build-arg MODD_VERSION=%s --build-arg GLIDE_VERSION=%s" % (build_args, buildInfo.modd, os.environ.get("GLIDE_VERSION"))

        if language == "java":
            build_args = "%s --build-arg MODD_VERSION=%s --build-arg JAVA_MAJOR_VERSION=%s" % (build_args, buildInfo.modd, version)
            build_context = "-f %s/Dockerfile.%s %s" % (language, version, language)
            run_command_exit('sed -e "s,{{JAVA_VERSION}},%s," %s/Dockerfile.tpl > %s/Dockerfile.%s' % (os.environ.get("JAVA_VERSION"), language, language, version), "fail to create Dockerfile for %s %s" % (language, os.environ.get("JAVA_VERSION")))

        if language == "node":
            build_args = "%s --build-arg MODD_VERSION=%s --build-arg NODE_VERSION=%s --build-arg NPM_VERSION=%s" % (build_args, buildInfo.modd, os.environ.get("NODE_VERSION"), os.environ.get("NPM_VERSION"))

        if language == "php":
            if version == "5.3":
                build_args = "%s --build-arg MODD_VERSION=%s --build-arg PHP_VERSION=%s --build-arg REDIS_VERSION=%s --build-arg SECURITY_CHECKER_VERSION=%s" % (build_args, buildInfo.modd, os.environ.get("PHP_VERSION"), os.environ.get("REDIS_VERSION"), os.environ.get("SECURITY_CHECKER_VERSION"))
                build_context = "%s/%s" % (language, version)
            else:
                build_args = "%s --build-arg MODD_VERSION=%s --build-arg APCU_VERSION=%s --build-arg COMPOSER_VERSION=%s --build-arg REDIS_VERSION=%s --build-arg SECURITY_CHECKER_VERSION=%s --build-arg XDEBUG_VERSION=%s" % (build_args, buildInfo.modd, os.environ.get("APCU_VERSION"), os.environ.get("COMPOSER_VERSION"), os.environ.get("REDIS_VERSION"), os.environ.get("SECURITY_CHECKER_VERSION"), os.environ.get("XDEBUG_VERSION"))
                build_context = "-f %s/Dockerfile.%s %s" % (language, version, language)
                run_command_exit('sed -e "s,{{PHP_VERSION}},%s," -e "s,{{PHP_MAJOR_VERSION}},%s," %s/Dockerfile.tpl > %s/Dockerfile.%s' % (os.environ.get("PHP_VERSION"), version.split(".")[0], language, language, version), "fail to create Dockerfile for %s %s" % (language, os.environ.get("PHP_VERSION")))

        if language == "python":
            build_args = "%s --build-arg PYTHON_VERSION=%s" % (
                build_args, os.environ.get("PYTHON_VERSION"))
            build_context = "-f %s/Dockerfile.%s %s" % (language, version, language)
            run_command_exit('sed -e "s,{{PYTHON_VERSION}},%s," -e "s,{{PYTHON_MAJOR_VERSION}},%s," %s/Dockerfile.tpl > %s/Dockerfile.%s' % (os.environ.get("PYTHON_VERSION"), version.split(".")[0], language, language, version), "fail to create Dockerfile for %s %s" % (language, os.environ.get("PYTHON_VERSION")))

        if language == "react-native":
            build_args = "%s --build-arg MODD_VERSION=%s" % (build_args, buildInfo.modd)

        if language == "sonar":
            build_args = "%s --build-arg GLIBC_VERSION=%s --build-arg SONARSCANNER_VERSION=%s" % (build_args, os.environ.get("GLIBC_VERSION"), os.environ.get("SONARSCANNER_VERSION"))

        if language == "sonarts":
            build_args = "%s --build-arg GLIBC_VERSION=%s --build-arg SONARSCANNER_VERSION=%s " % (build_args, os.environ.get("GLIBC_VERSION"), os.environ.get("SONARSCANNER_VERSION"))
            build_context = "-f %s/Dockerfile.%s %s" % (language, version, language)
            run_command_exit('sed -e "s,{{NODE_VERSION}},%s," %s/Dockerfile.tpl > %s/Dockerfile.%s' % (os.environ.get("NODE_VERSION"), language, language, version), "fail to create Dockerfile for %s %s" % (language, os.environ.get("VERSION")))



        cmd = "docker build -t %s %s --no-cache %s" % (image, build_args, build_context)

        print "> Run: %s" % cmd

        run_command_exit(cmd, "fail to build the image")

        run_command_exit("docker run %s %s ci-helper version -e" % (run_args, image), "Error with ci-helper installation")

        if language == "ansible":
            print "> Testing Ansible Image..."
            run_command_exit("docker run %s %s ansible --version" % (run_args, image),   "Error with ansible check")
            run_command_exit("docker run %s %s ansible-playbook --version" % (run_args, image), "Error with ansible-playbook check")

        if language == "aws":
            print "> Testing AWS Image..."
            run_command_exit("docker run %s %s aws --version" % (run_args, image), "Error with awscli check")
            run_command_exit("docker run %s %s python -c \"import boto3\"" % (run_args, image), "Error with boto3 check")
            run_command_exit("docker run %s %s python -c \"import yaml\"" % (run_args, image), "Error with PyYAML check")

        if language == "dind-aws":
            print "> Testing DIND - AWS Image..."
            run_command_exit("docker run %s %s aws --version" % (run_args, image), "Error with awscli check")
            run_command_exit("docker run %s %s docker --version" % (run_args, image), "Error with docker check (should be installed by docker image)")
            run_command_exit("docker run %s %s docker-compose --version" % (run_args, image), "Error with docker-compose check (should be installed by dind-aws image)")
            run_command_exit("docker run --privileged -d --name %s %s" % (language, image), "Error on starting dind-aws container")
            run_command_exit("docker exec %s docker ps" % (language), "Error with docker ps check")
            run_command_exit("docker rm -f %s" % (language), "Error on removing dind-aws container")

        if language == "golang":
            print "> Testing Golang Image..."
            run_command_exit("docker run %s %s aws --version" % (run_args, image),   "Error with awscli check")
            run_command_exit("docker run %s %s go version" % (run_args, image),      "Error with go check")
            run_command_exit("docker run %s %s glide --version" % (run_args, image), "Error with glide check")
            run_command_exit("docker run %s %s gin --version" % (run_args, image),   "Error with gin check")
            run_command_exit("docker run %s %s modd --version" % (run_args, image),  "Error with modd check")

        if language == "java":
            print "> Testing Java Image..."
            run_command_exit("docker run %s %s java -version" % (run_args, image), "Error with java check")
            run_command_exit("docker run %s %s mvn --version" % (run_args, image), "Error with mvn check")
            run_command_exit("docker run %s %s modd --version" % (run_args, image), "Error with modd check")

        if language == "node":
            print "> Testing Node Image..."
            run_command_exit("docker run %s %s node --version" % (run_args, image), "Error with node check")
            run_command_exit("docker run %s %s npm --version" % (run_args, image), "Error with npm check")
            run_command_exit("docker run %s %s sass --version" % (run_args, image), "Error with sass check")
            run_command_exit("docker run %s %s modd --version" % (run_args, image), "Error with modd check")

        if language == "php":
            print "> Testing PHP Image..."
            run_command_exit("docker run %s %s php --version" % (run_args, image), "Error with php check")
            run_command_exit("docker run %s %s composer --version" % (run_args, image), "Error with composer check")
            run_command_exit("docker run %s %s modd --version" % (run_args, image), "Error with modd check")
            run_command_exit("docker run %s %s security-checker --version" % (run_args, image), "Error with security-checker check")
            run_command_exit("docker run %s -v $(pwd)/php:/tmp %s php /tmp/test.php" % (run_args, image), "Error on testing php container")

        if language == "ruby":
            print "> Testing Ruby Image..."
            run_command_exit("docker run %s %s ruby --version" % (run_args, image),   "Error with ruby check")
            run_command_exit("docker run %s %s bundle --version" % (run_args, image), "Error with bundle check")

        if language == "react-native":
            print "> Testing React Native Image..."
            run_command_exit("docker run %s %s node --version" % (run_args, image), "Error with node check")
            run_command_exit("docker run %s %s npm --version" % (run_args, image), "Error with npm check")
            run_command_exit("docker run %s %s yarn --version" % (run_args, image), "Error with sass check")
            run_command_exit("docker run %s %s java -version" % (run_args, image), "Error with java check")
            run_command_exit("docker run %s %s react-native --version" % (run_args, image), "Error with java check")
            run_command_exit("docker run %s %s watchman --version" % (run_args, image), "Error with mvn check")
            run_command_exit("docker run %s %s rnpm --version" % (run_args, image), "Error with mvn check")
            run_command_exit("docker run %s %s modd --version" % (run_args, image), "Error with modd check")

        if language == "sonar":
            print "> Testing Sonar Scanner Image..."
            run_command_exit("docker run %s %s java -version" % (run_args, image),   "Error with java version")
            run_command_exit("docker run %s %s sonar-scanner -v" % (run_args, image), "Error with sonar-scanner check")

        if language == "sonarts":
            print "> Testing SonarTS Scanner Image..."
            run_command_exit("docker run %s %s node --version" % (run_args, image), "Error with node check")
            run_command_exit("docker run %s %s sonar-scanner -v" % (run_args, image), "Error with sonar-scanner check")

        print ""
        print "You can now test the image with the following command:\n   $ docker run --rm -ti %s" % image

    if push_image:
        run_command_exit("docker login --username %s --password %s" % (os.environ.get('DOCKER_USERNAME'), os.environ.get('DOCKER_PASSWORD')), "unable to login to docker")
        run_command_exit("docker push %s" % image, "unable to login to docker")


if __name__ == "__main__":
    main()
